src/Hl7.Fhir.Patch/Internal/ElementNodeAdapter.cs

```diff
@@ -0,0 +1,174 @@
+            {
+                errorMessage = $"The property at path '{name}' is already set. Use 'replace' operation instead.";
+                return false;
+            }
+
+            if (!TryConvertValue(value, PropertyDefinition, out var convertedValue))
+            {
+                errorMessage = $"The value '{value}' is invalid for target location.";
+                return false;
+            }
+
+            try
+            {
+                typedElement.Add(contractResolver, convertedValue, name);
+                errorMessage = null;
+                return true;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+                return false;
+            }
+        }
+
+        public virtual bool TryInsert (
+            object target,
+            int index,
+            IStructureDefinitionSummaryProvider contractResolver,
+            object value,
+            out string errorMessage)
+        {
+            errorMessage = $"The 'insert' operation can only be applied on the collection.";
+            return false;
+        }
+
+        public virtual bool TryDelete (
+            object target,
+            IStructureDefinitionSummaryProvider contractResolver,
+            out string errorMessage)
+        {
+            try
+            {
+                var typedElement = (ElementNode) target;
+                typedElement.Parent.Remove(typedElement);
+            }
+            catch ( Exception ex )
+            {
+                errorMessage = ex.Message;
+                return false;
+            }
+
+            errorMessage = null;
+            return true;
+        }
+
+        public virtual bool TryReplace(
+            object target,
+            IStructureDefinitionSummaryProvider contractResolver,
+            object value,
+            out string errorMessage)
+        {
+            var typedElement = (ElementNode) target;
+
+            if ( !TryConvertValue(value, typedElement.Definition, out var convertedValue) )
+            {
+                errorMessage = $"The value '{value}' is invalid for target location.";
+                return false;
+            }
+
+            try
+            {
+                var newElement = ElementNode.FromElement(convertedValue);
+                typedElement.Parent.Replace(contractResolver, typedElement, newElement);
+            }
+            catch ( Exception ex )
+            {
+                errorMessage = ex.Message;
+                return false;
+            }
+
+            errorMessage = null;
+            return true;
+        }
+
+        public virtual bool TryGet (
+            object target,
+            IStructureDefinitionSummaryProvider contractResolver,
+            out object value,
+            out string errorMessage)
+        {
+            value = target;
+            errorMessage = null;
+            return true;
+        }
+
+        protected virtual bool TryGetProperty(
```

Comment:
Naming: In general we use 'Element' when working with ITypedElement and 'Property' when working with POCO's.